### PR TITLE
[#3888] [#2704] Feature/ personal_sign and CallPrivateRPC methods support

### LINKED ISSUE #3888: Switch internal RPC calls to use CallPrivateRPC
### User Story

As a developer, I don't want to expose all JSON-RPC methods to DApps.

### Description

*Type*: Feature

*Summary*: In status-go, starting from commit `ef160e87`, we stripped down APIs exposed in `CallRPC`. So far, `CallRPC` could call **any** JSON-RPC method, even not whitelisted ones. 

In `ef160e87`, we fixed that and `CallRPC` calls only public or whitelisted APIs. In order to call a private API (like `admin_*`), a new binding called `CallPrivateRPC` must be used.

#### Expected behavior
`CallRPC` can't call non-whitelisted APIs.

#### Actual behavior
`CallRPC` can call **any** JSON-RPC method.

### Solution
*Summary*: 

- [ ] change `CallRPC` to `CallPrivateRPC` in internal usage (e.g. I remember we use `admin_peers` to check if MailServer is properly connected; after upgrading status-go to `ef160e87` it won't work with `CallRPC`),
- [ ] upgrade status-go to `ef160e87`.

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: any including 0.9.17
* Operating System: Android and iOS

### Related changes

- https://github.com/status-im/status-go/pull/815
- https://github.com/status-im/status-go/pull/825

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #3888, #2704

### CODE REVIEW

File: src/status_im/ui/screens/wallet/send/events.cljs [hunk 1]
```diff
@@ -195,8 +208,9 @@
                    (update-in [:wallet :transactions-unsigned] dissoc id)
                    (update-in [:wallet :send-transaction] merge clear-send-properties))}
           (if modal?
-            {:dispatch       [:navigate-back]
-             :dispatch-later [{:ms 400 :dispatch [:navigate-to-modal :wallet-transaction-sent-modal]}]}
+            (cond-> {:dispatch [:navigate-back]}
+                    (= method constants/web3-send-transaction)
+                    (assoc :dispatch-later [{:ms 400 :dispatch [:navigate-to-modal :wallet-transaction-sent-modal]}]))
```
Comment (janherich) [inline src/status_im/ui/screens/wallet/send/events.cljs:213]: I know that it was not introduced here, but why this delayed dispatch ?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/send/events.cljs:213]: idk maybe this is how modal window works, we need a delay between close and open modal windows

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/send/events.cljs:213]: i just remember it  worked wrong without delay

**Review by jeluard — APPROVED** [review]

**Review by goranjovic — APPROVED** [review]

**Comment by goranjovic** [comment]
@flexsurfer have a screenshot?

**Comment by flexsurfer** [comment]
@goranjovic do you mean this 

image

**Review by goranjovic — APPROVED** [review]

**Comment by asemiankevich** [comment]
image

image

image

image

**Pull request merged**